include/stomp_moveit/filter_functions.hpp

```diff
@@ -23,5 +24,32 @@ FilterFn simple_smoothing_matrix(size_t num_timesteps)
     return true;
   };
 }
+FilterFn enforce_position_bounds(const moveit::core::JointModelGroup* group)
+{
+  return [=](const Eigen::MatrixXd& values, Eigen::MatrixXd& filtered_values) {
```

Comment:
in general, to prevent dangling references. Here, the group pointer is a temporary copy, returning the lambda with reference capture would lead to undefined behavior, possibly segfaults.